pyproject.toml

```
@@ -1,9 +1,57 @@
 [build-system]
-requires = [
-    "setuptools >= 46.4.0",
-]
+requires = ["setuptools >= 70.0.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name        = "aiohttp"
+license     = {text = "Apache-2.0"}
+description = "Async http client/server framework (asyncio)"
+readme      = "README.rst"
+maintainers = [{name = "aiohttp team", email = "team@aiohttp.org"}]
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Framework :: AsyncIO",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: Apache Software License",
+  "Operating System :: POSIX",
+  "Operating System :: MacOS :: MacOS X",
+  "Operating System :: Microsoft :: Windows",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Internet :: WWW/HTTP",
+]
+requires-python = ">=3.9"
+dynamic = ["version", "dependencies", "optional-dependencies"]
+
+[project.urls]
+"Homepage"           = "https://github.com/aio-libs/aiohttp"
+"Chat: Matrix"       = "https://matrix.to/#/#aio-libs:matrix.org"
+"Chat: Matrix Space" = "https://matrix.to/#/#aio-libs-space:matrix.org"
+"CI: GitHub Actions" = "https://github.com/aio-libs/aiohttp/actions?query=workflow%3ACI"
+"Coverage: codecov"  = "https://codecov.io/github/aio-libs/aiohttp"
+"Docs: Changelog"    = "https://docs.aiohttp.org/en/stable/changes.html"
+"Docs: RTD"          = "https://docs.aiohttp.org"
+"GitHub: issues"     = "https://github.com/aio-libs/aiohttp/issues"
+"GitHub: repo"       = "https://github.com/aio-libs/aiohttp"
+
+[tool.setuptools.dynamic]
+version = {attr = "aiohttp.__version__"}
+
+[tool.setuptools.packages.find]
+include = ["aiohttp*"]
```

This will probably happen automatically.

---

Unfortunately not. Setuptools will throw an error `Multiple top-level packages discovered in a flat-layout: ['vendor', 'aiohttp', 'CHANGES']`.

---

Is the wildcard needed?
```suggestion
include = [
  "aiohttp",
]
```
Or do we need something like this?
```suggestion
include = [
  "aiohttp",
  "aiohttp.*",
]
```

Personally, though, I'd prefer exclusions over inclusions here.

---

> Is the wildcard needed?

Technically no since all subpackages would be included as package data anyway. However, I usually still recommend it just in case the package data include / exclude changes.

> Or do we need something like this? [`"aiohttp"`, `"aiohttp.*"`]

Not sure there is an advantage using it over just `aiohttp*`.

> Personally, though, I'd prefer exclusions over inclusions here.

Exclusion is quite easy to get accidentally wrong. Just `exclude = ["examples"]` as it was before wouldn't be enough. Any folder included in the sdist which contains at least one `.py` file would be included / have to be manually excluded (i.e. `docs`, `requirements`, `tests`, `tools`, `vendor`).

--
Overall I'd recommend to stick with `include = ["aiohttp*"]` in this case.

---

Go with the second code suggestion, then. I don't like that `aiohttp*` would match something that may get in accidentally, like `aiohttp1`.

---

Done. Added your code suggestion. Also double checked it again locally. The `RECORD` file still looks good.